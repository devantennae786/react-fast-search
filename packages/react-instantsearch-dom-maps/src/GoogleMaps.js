import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { createClassNames } from 'react-instantsearch-dom';
import { LatLngPropType, BoundingBoxPropType } from './propTypes';

const cx = createClassNames('GeoSearch');

class GoogleMaps extends Component {
  static propTypes = {
    google: PropTypes.object.isRequired,
    initialZoom: PropTypes.number.isRequired,
    initialPosition: LatLngPropType.isRequired,
    mapOptions: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    onIdle: PropTypes.func.isRequired,
    shouldUpdate: PropTypes.func.isRequired,
    boundingBox: BoundingBoxPropType,
    boundingBoxPadding: PropTypes.number,
    children: PropTypes.node,
  };

  static childContextTypes = {
    // eslint-disable-next-line @typescript-eslint/camelcase
    __ais_geo_search__google_maps__: PropTypes.shape({
      google: PropTypes.object,
      instance: PropTypes.object,
    }),
  };

  isUserInteraction = true;
  isPendingRefine = false;
  listeners = [];

  state = {
    isMapReady: false,
  };

  getChildContext() {
    const { google } = this.props;

    return {
      // eslint-disable-next-line @typescript-eslint/camelcase
      __ais_geo_search__google_maps__: {
        instance: this.instance,
        google,
      },
    };
  }

  componentDidMount() {
    const { google, mapOptions } = this.props;

    this.instance = new google.maps.Map(this.element, {
      mapTypeControl: false,
      fullscreenControl: false,
      streetViewControl: false,
      clickableIcons: false,
      zoomControlOptions: {
        position: google.maps.ControlPosition.LEFT_TOP,
      },
      ...mapOptions,
    });

    this.listeners.push(
      google.maps.event.addListenerOnce(
        this.instance,
        'idle',
        this.setupListenersWhenMapIsReady
      )
    );
  }

  componentDidUpdate() {
    const {
      google,
      initialZoom,
      initialPosition,
      boundingBox,
      boundingBoxPadding,
      shouldUpdate,
    } = this.props;

    if (!shouldUpdate()) {
      return;
    }

    if (boundingBox) {
      this.lockUserInteration(() => {
        this.instance.fitBounds(
          new google.maps.LatLngBounds(
            boundingBox.southWest,
            boundingBox.northEast
          ),
          boundingBoxPadding
        );
      });
    } else {
      this.lockUserInteration(() => {
        this.instance.setZoom(initialZoom);
        this.instance.setCenter(initialPosition);
      });
    }
  }

  componentWillUnmount() {
    this.listeners.forEach(listener => {
      listener.remove();
    });

    this.listeners = [];
  }

  setupListenersWhenMapIsReady = () => {
    this.listeners = [];

    this.setState(() => ({
      isMapReady: true,
    }));

    const onChange = () => {
      if (this.isUserInteraction) {
        this.props.onChange();
      }
    };

    this.listeners.push(this.instance.addListener('center_changed', onChange));
    this.listeners.push(this.instance.addListener('zoom_changed', onChange));
    this.listeners.push(this.instance.addListener('dragstart', onChange));

    this.listeners.push(
      this.instance.addListener('idle', () => {
        if (this.isUserInteraction) {
          this.props.onIdle({
            instance: this.instance,
          });
        }
      })
    );
  };

  lockUserInteration(functionThatAltersTheMapPosition) {
    this.isUserInteraction = false;
    functionThatAltersTheMapPosition();
    this.isUserInteraction = true;
  }

  render() {
    const { children } = this.props;
    const { isMapReady } = this.state;

    return (
      <div className={cx('')}>
        <div ref={c => (this.element = c)} className={cx('map')} />
        {isMapReady && children}
      </div>
    );
  }
}

export default GoogleMaps;
