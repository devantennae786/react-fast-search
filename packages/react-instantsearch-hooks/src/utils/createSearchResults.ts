import algoliasearchHelper from 'algoliasearch-helper';

import type { SearchParameters } from 'algoliasearch-helper';

export function createSearchResults(state: SearchParameters) {
  return new algoliasearchHelper.SearchResults(state, [
    {
      query: state.query ?? '',
      page: state.page ?? 0,
      hitsPerPage: state.hitsPerPage ?? 20,
      hits: [],
      nbHits: 0,
      nbPages: 0,
      params: '',
      exhaustiveNbHits: true,
      exhaustiveFacetsCount: true,
      processingTimeMS: 0,
      index: state.index,
    },
  ]);
}
