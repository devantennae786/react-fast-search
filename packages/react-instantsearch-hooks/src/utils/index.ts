export * from './invariant';
export * from './noop';
export * from './warn';
