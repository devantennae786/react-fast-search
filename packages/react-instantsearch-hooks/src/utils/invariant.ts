/**
 * Throws an error if the condition is not met in development mode.
 * This is used to make development a better experience to provide guidance as
 * to where the error comes from.
 */
export function invariant(
  condition: boolean,
  message: string | (() => string)
) {
  if (condition) {
    return;
  }

  if (!__DEV__) {
    throw new Error('Invariant failed');
  }

  if (__DEV__) {
    throw new Error(
      `[InstantSearch] ${typeof message === 'function' ? message() : message}`
    );
  }
}
