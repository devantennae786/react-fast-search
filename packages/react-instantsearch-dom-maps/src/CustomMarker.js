import { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import createHTMLMarker from './elements/createHTMLMarker';
import { registerEvents, createListenersPropTypes } from './utils';
import { GeolocHitPropType } from './propTypes';
import { GOOGLE_MAPS_CONTEXT } from './GoogleMaps';

const eventTypes = {
  onClick: 'click',
  onDoubleClick: 'dblclick',
  onMouseDown: 'mousedown',
  onMouseEnter: 'mouseenter',
  onMouseLeave: 'mouseleave',
  onMouseMove: 'mousemove',
  onMouseOut: 'mouseout',
  onMouseOver: 'mouseover',
  onMouseUp: 'mouseup',
};

class CustomMarker extends Component {
  static propTypes = {
    ...createListenersPropTypes(eventTypes),
    hit: GeolocHitPropType.isRequired,
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    anchor: PropTypes.shape({
      x: PropTypes.number.isRequired,
      y: PropTypes.number.isRequired,
    }),
  };

  static contextTypes = {
    [GOOGLE_MAPS_CONTEXT]: PropTypes.shape({
      google: PropTypes.object,
      instance: PropTypes.object,
    }),
  };

  static defaultProps = {
    className: '',
    anchor: {
      x: 0,
      y: 0,
    },
  };

  static isReact16() {
    return typeof ReactDOM.createPortal === 'function';
  }

  state = {
    marker: null,
  };

  componentDidMount() {
    const { hit, className, anchor } = this.props;
    const { google, instance } = this.context[GOOGLE_MAPS_CONTEXT];
    // Not the best way to create the reference of the CustomMarker
    // but since the Google object is required didn't find another
    // solution. Ideas?
    const Marker = createHTMLMarker(google);

    const marker = new Marker({
      map: instance,
      position: hit._geoloc,
      className,
      anchor,
    });

    this.removeListeners = registerEvents(eventTypes, this.props, marker);

    this.setState(() => ({
      marker,
    }));
  }

  componentDidUpdate() {
    const { children } = this.props;
    const { marker } = this.state;

    this.removeListeners();

    this.removeListeners = registerEvents(eventTypes, this.props, marker);

    if (!CustomMarker.isReact16()) {
      ReactDOM.unstable_renderSubtreeIntoContainer(
        this,
        children,
        marker.element
      );
    }
  }

  componentWillUnmount() {
    const { marker } = this.state;

    if (!CustomMarker.isReact16()) {
      ReactDOM.unmountComponentAtNode(marker.element);
    }

    marker.setMap(null);
  }

  render() {
    const { children } = this.props;
    const { marker } = this.state;

    if (!marker || !CustomMarker.isReact16()) {
      return null;
    }

    return ReactDOM.createPortal(children, marker.element);
  }
}

export default CustomMarker;
