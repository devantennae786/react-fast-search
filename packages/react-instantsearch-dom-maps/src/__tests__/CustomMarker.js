import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  createFakeGoogleReference,
  createFakeMapInstance,
  createFakeHTMLMarkerInstance,
} from '../../test/mockGoogleMaps';
import createHTMLMarker from '../elements/createHTMLMarker';
import * as utils from '../utils';
import { GOOGLE_MAPS_CONTEXT } from '../GoogleMaps';
import CustomMarker from '../CustomMarker';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../elements/createHTMLMarker', () => jest.fn());

jest.mock('../utils');

describe('CustomMarker', () => {
  const defaultProps = {
    hit: {
      _geoloc: {
        lat: 10,
        lng: 12,
      },
    },
  };

  beforeEach(() => {
    utils.registerEvents.mockClear();
    utils.registerEvents.mockReset();

    // Register default implementation
    utils.registerEvents.mockImplementation(() => jest.fn());
  });

  describe('creation', () => {
    it('expect to create the marker on didMount with default options', () => {
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      const wrapper = shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(createHTMLMarker).toHaveBeenCalledWith(google);
      expect(wrapper.state('marker')).toBe(marker);

      expect(factory).toHaveBeenCalledTimes(1);
      expect(factory).toHaveBeenCalledWith(
        expect.objectContaining({
          map: mapInstance,
          position: {
            lat: 10,
            lng: 12,
          },
        })
      );
    });

    it('expect to create the marker on didMount with given options', () => {
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
        className: 'my-marker',
        anchor: {
          x: 10,
          y: 10,
        },
      };

      shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(factory).toHaveBeenCalledWith(
        expect.objectContaining({
          className: 'my-marker',
          anchor: {
            x: 10,
            y: 10,
          },
        })
      );
    });

    it('expect to register the listeners on didMount', () => {
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(utils.registerEvents).toHaveBeenCalledTimes(2); // cDM + cDU
      expect(utils.registerEvents).toHaveBeenCalledWith(
        expect.any(Object),
        expect.any(Object),
        marker
      );
    });
  });

  describe('update', () => {
    it('expect to remove the listeners on didUpdate', () => {
      const removeEventListeners = jest.fn();
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      utils.registerEvents.mockImplementation(() => removeEventListeners);

      const props = {
        ...defaultProps,
      };

      shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(removeEventListeners).toHaveBeenCalledTimes(1);
    });

    it('expect to register the listeners on didUpdate', () => {
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(utils.registerEvents).toHaveBeenCalledTimes(2); // cDM + cDU
      expect(utils.registerEvents).toHaveBeenCalledWith(
        expect.any(Object),
        expect.any(Object),
        marker
      );
    });
  });

  describe('delete', () => {
    it('expect to remove the Marker on willUnmount', () => {
      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      const wrapper = shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      wrapper.unmount();

      expect(marker.setMap).toHaveBeenCalledTimes(1);
      expect(marker.setMap).toHaveBeenCalledWith(null);
    });
  });

  describe('with portal', () => {
    it('expect to render correctly', () => {
      const unstableRenderSubtreeIntoContainer = jest.spyOn(
        ReactDOM,
        'unstable_renderSubtreeIntoContainer'
      );

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      utils.registerEvents.mockImplementation(() => () => {});

      const props = {
        ...defaultProps,
      };

      // Use `mount` instead of `shallow` to trigger the render
      // of createPortal otherwise the Snapshot is empty
      const wrapper = mount(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(wrapper).toMatchSnapshot();

      expect(unstableRenderSubtreeIntoContainer).not.toHaveBeenCalled();

      unstableRenderSubtreeIntoContainer.mockReset();
      unstableRenderSubtreeIntoContainer.mockRestore();
    });

    it('expect to render correctly without a marker', () => {
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      const props = {
        ...defaultProps,
      };

      const wrapper = shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          disableLifecycleMethods: true,
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(wrapper.type()).toBe(null);
    });

    it('expect to not render on didUpdate', () => {
      const unstableRenderSubtreeIntoContainer = jest.spyOn(
        ReactDOM,
        'unstable_renderSubtreeIntoContainer'
      );

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      utils.registerEvents.mockImplementation(() => () => {});

      const props = {
        ...defaultProps,
      };

      // Use `mount` instead of `shallow` to trigger didUpdate
      const wrapper = mount(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(unstableRenderSubtreeIntoContainer).not.toHaveBeenCalled();

      wrapper.instance().componentDidUpdate();

      expect(unstableRenderSubtreeIntoContainer).not.toHaveBeenCalled();

      unstableRenderSubtreeIntoContainer.mockReset();
      unstableRenderSubtreeIntoContainer.mockRestore();
    });

    it('expect to not call unmountComponentAtNode on willUnmount', () => {
      const unmountComponentAtNode = jest.spyOn(
        ReactDOM,
        'unmountComponentAtNode'
      );

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      const wrapper = shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      wrapper.unmount();

      expect(unmountComponentAtNode).not.toHaveBeenCalled();

      unmountComponentAtNode.mockReset();
      unmountComponentAtNode.mockRestore();
    });
  });

  describe('with unstable_renderSubtreeIntoContainer', () => {
    it('expect to render correctly', () => {
      const unstableRenderSubtreeIntoContainer = jest.spyOn(
        ReactDOM,
        'unstable_renderSubtreeIntoContainer'
      );

      const isReact16 = jest
        .spyOn(CustomMarker, 'isReact16')
        .mockImplementation(() => false);

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      utils.registerEvents.mockImplementation(() => () => {});

      const props = {
        ...defaultProps,
      };

      const wrapper = mount(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(wrapper).toMatchSnapshot();

      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledTimes(1);
      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledWith(
        wrapper.instance(),
        <span>This is the children.</span>,
        marker.element
      );

      unstableRenderSubtreeIntoContainer.mockReset();
      unstableRenderSubtreeIntoContainer.mockRestore();

      isReact16.mockReset();
      isReact16.mockRestore();
    });

    it('expect to render correctly without a marker', () => {
      const isReact16 = jest
        .spyOn(CustomMarker, 'isReact16')
        .mockImplementation(() => false);

      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      const props = {
        ...defaultProps,
      };

      const wrapper = shallow(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          disableLifecycleMethods: true,
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(wrapper.type()).toBe(null);

      isReact16.mockReset();
      isReact16.mockRestore();
    });

    it('expect to render on didUpdate', () => {
      const unstableRenderSubtreeIntoContainer = jest.spyOn(
        ReactDOM,
        'unstable_renderSubtreeIntoContainer'
      );

      const isReact16 = jest
        .spyOn(CustomMarker, 'isReact16')
        .mockImplementation(() => false);

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      utils.registerEvents.mockImplementation(() => () => {});

      const props = {
        ...defaultProps,
      };

      // Use `mount` instead of `shallow` to avoid issue with `unstable_renderSubtreeIntoContainer`
      const wrapper = mount(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledTimes(1);
      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledWith(
        wrapper.instance(),
        <span>This is the children.</span>,
        marker.element
      );

      wrapper.instance().componentDidUpdate();

      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledTimes(2);
      expect(unstableRenderSubtreeIntoContainer).toHaveBeenCalledWith(
        wrapper.instance(),
        <span>This is the children.</span>,
        marker.element
      );

      unstableRenderSubtreeIntoContainer.mockReset();
      unstableRenderSubtreeIntoContainer.mockRestore();

      isReact16.mockReset();
      isReact16.mockRestore();
    });

    it('expect to call unmountComponentAtNode on willUnmount', () => {
      const unmountComponentAtNode = jest.spyOn(
        ReactDOM,
        'unmountComponentAtNode'
      );

      const isReact16 = jest
        .spyOn(CustomMarker, 'isReact16')
        .mockImplementation(() => false);

      const marker = createFakeHTMLMarkerInstance();
      const factory = jest.fn(() => marker);
      const mapInstance = createFakeMapInstance();
      const google = createFakeGoogleReference({
        mapInstance,
      });

      createHTMLMarker.mockImplementationOnce(() => factory);

      const props = {
        ...defaultProps,
      };

      // Use `mount` instead of `shallow` to avoid issue with `unstable_renderSubtreeIntoContainer`
      const wrapper = mount(
        <CustomMarker {...props}>
          <span>This is the children.</span>
        </CustomMarker>,
        {
          context: {
            [GOOGLE_MAPS_CONTEXT]: {
              instance: mapInstance,
              google,
            },
          },
        }
      );

      wrapper.unmount();

      expect(unmountComponentAtNode).toHaveBeenCalledWith(marker.element);

      unmountComponentAtNode.mockReset();
      unmountComponentAtNode.mockRestore();

      isReact16.mockReset();
      isReact16.mockRestore();
    });
  });
});
