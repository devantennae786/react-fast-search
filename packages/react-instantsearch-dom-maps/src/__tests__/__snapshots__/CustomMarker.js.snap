// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMarker with portal expect to render correctly 1`] = `
<CustomMarker
  anchor={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  className=""
  hit={
    Object {
      "_geoloc": Object {
        "lat": 10,
        "lng": 12,
      },
    }
  }
>
  <Portal
    containerInfo={
      <div>
        <span>
          This is the children.
        </span>
      </div>
    }
  >
    <span>
      This is the children.
    </span>
  </Portal>
</CustomMarker>
`;

exports[`CustomMarker with unstable_renderSubtreeIntoContainer expect to render correctly 1`] = `
<CustomMarker
  anchor={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  className=""
  hit={
    Object {
      "_geoloc": Object {
        "lat": 10,
        "lng": 12,
      },
    }
  }
/>
`;
